pkg/crc/machine/machine.go

```diff
@@ -328,13 +330,59 @@ func Ip(ipConfig IpConfig) (IpResult, error) {
 		result.Error = err.Error()
 		return *result, err
 	}
-
 	if result.IP, err = host.Driver.GetIP(); err != nil {
 		result.Success = false
 		result.Error = err.Error()
 		return *result, err
 	}
+	return *result, nil
+}
+
+func Status(statusConfig ClusterStatusConfig) (ClusterStatusResult, error) {
+	result := &ClusterStatusResult{Name: statusConfig.Name, Success: true}
+	api := libmachine.NewClient(constants.MachineBaseDir, constants.MachineCertsDir)
+	_, err := api.Exists(statusConfig.Name)
+	if err != nil {
+		result.Success = false
+		result.Error = err.Error()
+		return *result, err
+	}
+	openshiftStatus := "Stopped"
+	diskUsage := "Unknown"
 
+	host, err := api.Load(statusConfig.Name)
+	if err != nil {
+		result.Success = false
+		result.Error = err.Error()
+		return *result, err
+	}
+	vmStatus, err := host.Driver.GetState()
+	if err != nil {
+		result.Success = false
+		result.Error = err.Error()
+		return *result, err
+	}
+
+	if vmStatus == state.Running {
+		// check if all the clusteroperators are running
+		ocConfig := oc.UseOCWithConfig(statusConfig.Name)
+		cosAvailable, err := oc.GetClusterOperatorStatus(ocConfig)
+		if err != nil {
+			result.Success = false
+			result.Error = err.Error()
+			return *result, err
+		}
+		if cosAvailable {
+			// TODO:get openshift version as well and add to status
+			openshiftStatus = fmt.Sprintf("Running (v4.x)")
+		}
+
+		diskSize, diskUse, mountpoint := getDiskUsage(host.Driver, "/dev/vda3")
+		diskUsage = fmt.Sprintf("%s of %s (Mounted On: %s)", diskUse, diskSize, mountpoint)
```

Comment:
Do we really need the mountpoint?
I'd prefer that machine.Status returns the raw data, and then `cmd/crc/cmd/status.go` will do the string formatting. This means changing `ClusterStatusResult` to have `DiskUse int64; DiskSize int64; MountPoint string;` (pseudo code)

Reply:
I did not modify this function, this is taken from the minishift code base, will change it accordingly